Frontend/src/hooks/navigation.ts

```diff
@@ -1,10 +1,11 @@
-import { useRef, useEffect } from 'react';
+import React, { useRef, useEffect } from 'react';
 
 export function useOnClickOutside(
   onClickOutside: () => void
 ): React.RefObject<HTMLDivElement> {
   const ref = useRef<HTMLDivElement>(null);
   useEffect(() => {
+    // eslint-disable-next-line
     const handleClickOutside = (event: any) => {
```

Comment:
```suggestion
    const handleClickOutside:  EventHandler<MouseEvent> = (event) => {
```

Kanskje dette funker

Reply:
Hos meg klager typescript da med:  `TS2344: Type 'MouseEvent' does not satisfy the constraint 'SyntheticEvent<any, Event>'.` og `TS2345: Argument of type 'EventTarget | null' is not assignable to parameter of type 'Node | null'.` på linjen under.

Men dette ser ut til å funke:
```typescript
const handleClickOutside = (event: MouseEvent) => {
  if (event.target instanceof HTMLElement && ref.current?.contains(event.target)) {
    ...
  )
  ...
};
```

Har ikke testet det funksjonelt sett, bare sett at Typescript blir happy 🙈